python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -1322,6 +1322,12 @@ def test_table_batch_empty_dataframe(self):
         _check_pandas_roundtrip(df2, preserve_index=True)
         _check_pandas_roundtrip(df2, as_batch=True, preserve_index=True)
 
+    def test_convert_empty_table(self):
+        arr = pa.array([], type=pa.int64())
+        npt.assert_array_equal(arr.to_pandas(), np.array([]))
+        arr = pa.array([], type=pa.string())
+        npt.assert_array_equal(arr.to_pandas(), np.array([]))
```

Comment:
`np.array([])` returns `float64` for the dtype, is that expected?

Reply:
I skipped adding the `dtype` here as `assert_array_equal` seems to ignore it anyways. I can add it if it's better for clarification.

Reply:
Can we test dtype explicitly as well?

Reply:
you should use the pandas versions of these which work well